compiler/rustc_middle/src/mir/visit.rs

```diff
@@ -1394,9 +1394,8 @@ impl PlaceContext {
     pub fn is_borrow(self) -> bool {
```

Comment:
Might be worth updating the doc comment to explicitly say that only "real" borrows (that exist operationally) are considered here?

Another option might be to move the packed-place-borrow-checker to after SharedBorrow deletion, but that might be a bit too late.

Reply:
Yeah, I presumed that the packed checker was early on so as to avoid any optimizations getting in the way, though we shouldn't be applying any semantics-affecting optimizations until *after* borrowck, so perhaps this could be moved to right after post-borrowck cleanup 🤔

Reply:
I'll change the docs anyways.

Reply:
Do fake borrows get removed somewhere within analysis MIR? As ling as we can stay on analysis MIR, it should be fine, that one should not be affected by optimizations at all, right?